testkit/src/main/scala/org/bitcoins/testkit/wallet/BitcoinSWalletTest.scala

```diff
@@ -52,7 +52,8 @@ trait BitcoinSWalletTest
 
   def destroyWallet(wallet: UnlockedWalletApi): Future[Unit] = {
     WalletDbManagement
-      .dropAll()(config = config.walletConf, ec = implicitly[ExecutionContext])
+      .dropAll()(config = wallet.walletConfig,
+                 ec = implicitly[ExecutionContext])
```

Comment:
This doesn't resolve to `ExecutionContext.global` unless that is explicitly imported. 

For all def with implicit params:

```scala
def foo(bar: Int)(implicit s: Something): Unit = ???
```

All of the below snippets are equivalent: 

```scala
implicit val s: Something = ???
foo(2) // implicitly passed
```

```scala
val s: Something = ???
foo(2)(s) // explicitly passed
```

```scala
implicit val s: Something = ???
val otherS: Something = implicitly[Something] // resolves to s above
foo(2)(otherS)
```

```scala
implicit  val s: Something = ???
foo(2)(implicitly[Something] // this resolves to s above
```

There's no point in the last snippet (because `foo` only takes one implicit param). However on the line you commented on, there's two implicit params. So we're basically saying: "use this specific wallet config, but conjure up the `ExecutionContext` following normal implicit resolution rules"

Reply:
We want to control precisely which wallet is being destroyed (which is achieved here by making sure we're operating on the correct datadir). If we're adding an implicit parameter to the method we're not getting any guarantees about what wallet we're destroying because it's just grabbing whatever implicit configuration is closest in scope.